packages/cloud_firestore/cloud_firestore/example/test_driver/load_bundle_e2e.dart

```diff
@@ -0,0 +1,108 @@
+// import 'package:test/test.dart';
+
+void runLoadBundleTests() {
+  group('$DocumentReference', () {
+    late FirebaseFirestore firestore;
+    final url = Uri.https('api.rnfirebase.io', '/firestore/bundle');
+    const String collection = 'firestore-bundle-tests';
+
+    Future<Uint8List> loadBundleSetup() async {
+      // endpoint serves a bundle with 3 documents each containing
+      // a 'number' property that increments in value 1-3.
+      final response = await http.get(url);
+      String string = response.body;
+      return Uint8List.fromList(string.codeUnits);
+    }
+
+    setUp(() {
+      firestore = FirebaseFirestore.instance;
+    });
+
+    group('FirebaseFirestore.loadBundle()', () {
+      test('loadBundle()', () async {
+        Uint8List buffer = await loadBundleSetup();
+        LoadBundleTask task = firestore.loadBundle(buffer);
+
+        // ensure the bundle has been completely cached
+        await task.stream().last;
+
+        QuerySnapshot<Map<String, Object?>> snapshot = await firestore
+            .collection(collection)
+            .get(const GetOptions(source: Source.cache));
+
+        expect(snapshot.docs.map((document) => document['number']),
+            everyElement(anyOf(1, 2, 3)));
+      });
+
+      test('loadBundle(): LoadBundleTaskProgress stream snapshots', () async {
+        Uint8List buffer = await loadBundleSetup();
+        LoadBundleTask task = firestore.loadBundle(buffer);
+
+        final list = await task.stream().toList();
+
+        expect(list.map((e) => e.totalDocuments), everyElement(isNonNegative));
+        expect(list.map((e) => e.bytesLoaded), everyElement(isNonNegative));
+        expect(list.map((e) => e.documentsLoaded), everyElement(isNonNegative));
+        expect(list.map((e) => e.totalBytes), everyElement(isNonNegative));
+        expect(list, everyElement(isInstanceOf<LoadBundleTaskSnapshot>()));
+        expect(
+            list.map((e) => e.taskState),
+            everyElement(anyOf(
+                LoadBundleTaskState.running, LoadBundleTaskState.success)));
+
+        LoadBundleTaskSnapshot lastSnapshot = list.last;
+        expect(lastSnapshot.taskState, LoadBundleTaskState.success);
+      });
+    });
+
+    group('FirebaeFirestore.namedQueryGet()', () {
+      test('namedQueryGet() successful', () async {
+        Uint8List buffer = await loadBundleSetup();
+        LoadBundleTask task = firestore.loadBundle(buffer);
+
+        // ensure the bundle has been completely cached
+        await task.stream().last;
+
+        //namedQuery 'named-bundle-test' which returns a QuerySnaphot of the same 3 documents
+        // with 'number' property
+        QuerySnapshot<Map<String, Object?>> snapshot =
+            await firestore.namedQueryGet('named-bundle-test',
+                options: const GetOptions(source: Source.cache));
+
+        expect(snapshot.docs.map((document) => document['number']),
+            everyElement(anyOf(1, 2, 3)));
```

Comment:
Same as before about unordered comparison

Reply:
as mentioned before, I'm not expecting an ordered list so I don't think it matters.